### PR TITLE
Fix student union room booking

### LINKED ISSUE #48: Booking for student union building rooms uses wrong url
At the moment of writing this TimeEdit for Chalmers uses two different versions for booking regular group rooms and rooms in the student union building, `chalmers_test` and `chalmers` respectivly`. However, the current implementation uses only `chalmers_test`. This causes a problem since even though we cannot see the possibility to book student union rooms on `chalmers_test` in the web browser the api still allows it. This allows for two bookings identical bookings one on each version which could lead to a lot of confusion.

### PR DESCRIPTION
**Pull request by HerrNaN**
Resolves #48

### CODE REVIEW

File: internal/booking/timeedit/bookingService.go [hunk 1]
```diff
@@ -11,26 +11,40 @@ import (
...
-const providerName = "ChalmersTimeEdit"
+const BaseProvider = "TimeEdit"
+
+var studentUnionRooms = []string{
+	"Grupprum 1",
+	"Grupprum 2",
+	"Grupprum 3",
+	"Musikrummet",
+	"Motionshallen",
+}
```
Comment (williamleven) [inline internal/booking/timeedit/bookingService.go:38]: Is there no filter we can use on the api instead? how does the timeedit frontend figure out which ones to show?

File: internal/booking/timeedit/bookingService.go [hunk 2]
```diff
@@ -11,26 +11,40 @@ import (
...
 
 type BookingService struct {
 	client *http.Client
-	rooms  rooms
+	rooms  Rooms
```
Comment (williamleven) [inline internal/booking/timeedit/bookingService.go:42]: why public

File: internal/booking/timeedit/bookingService.go [hunk 3]
```diff
@@ -11,26 +11,40 @@ import (
...
 }
 
-func (bs BookingService) Book(booking booking.Booking) error {
+func (bs BookingService) Book(booking booking.Booking, timeEditVersion string) error {
```
Comment (williamleven) [inline internal/booking/timeedit/bookingService.go:45]: Wouldn't it be more nice to have `timeEditVersion string` as a parameter to the constructor? Would save you a lot of complexity and code

**Review by williamleven — CHANGES_REQUESTED** [review]
I left some comments which i think could greatly simplify your solution. Nice that you got it working!

File: internal/booking/timeedit/bookingService.go (hunk 1 above)
Comment (HerrNaN) [reply internal/booking/timeedit/bookingService.go:38]: I don't really know how the timeedit frontend knows how, but we could indeed make a separate call to get only these rooms and use that call to fill this array. :+1:

File: internal/booking/timeedit/room.go [hunk 4]
```diff
@@ -28,3 +28,23 @@ func (rs rooms) nameFromId(id string) (string, error) {
...
+	}
+	return rs
+}
+
+func (rs rooms) removeMany(rooms rooms) rooms {
+	for _, n := range rooms {
+		rs = rs.remove(n)
+	}
+	return rs
+}
```
Comment (williamleven) [inline internal/booking/timeedit/room.go:50]: Looks like this code is dead?

**Review by williamleven — COMMENTED** [review]
Well done with the changes!

File: internal/booking/timeedit/room.go (hunk 4 above)
Comment (HerrNaN) [reply internal/booking/timeedit/room.go:50]: Nope it's used [here](https://github.com/sidusIO/BooGroCha/pull/50/files#diff-4caae089efd20f4253f987e704f0133aR287) in `getRooms()`

**Review by williamleven — APPROVED** [review]

File: internal/booking/timeedit/room.go (hunk 4 above)
Comment (HerrNaN) [reply internal/booking/timeedit/room.go:50]: Not that I could find

**Pull request merged**